src/api/chart-api.ts

```diff
@@ -236,7 +236,7 @@ export class ChartApi implements IChartApi, DataUpdatesConsumer<SeriesType> {
 		const timeScaleUpdate = update.timeScaleUpdate;
 		model.updateTimeScale(timeScaleUpdate.index, timeScaleUpdate.changes, timeScaleUpdate.marks, true);
 		timeScaleUpdate.seriesUpdates.forEach((value: SeriesUpdatePacket, key: Series) => {
-			key.updateData(value.update);
+			key.updateData(value.update, true);
```

Comment:
Why can't we call `clearData` method here instead of passing `true`? Also, looks like we don't need to call `clearData` for the series in [`DataLayer::setSeriesData`](https://github.com/tradingview/lightweight-charts/blob/master/src/api/data-layer.ts#L261)?

Reply:
Because `clearData` clears also pallete which was filled right before in `DataLayer`

Reply:
Oh crap 💩... Looks like we should fix it in https://github.com/tradingview/lightweight-charts/issues/270#issue-529280808 😂

Reply:
What do you think about something like that?

```suggestion
			// the latest arg `true` must be removed in https://github.com/tradingview/lightweight-charts/issues/270
			// here we don't need to clear palettes because they were just filled in DataLayer
			// see https://github.com/tradingview/lightweight-charts/pull/330#discussion_r379415805
			key.updateData(value.update, true);
```

Reply:
Looks good